adapter/src/dbt/adapters/fal/wrappers.py

```diff
@@ -41,6 +45,18 @@ def __init__(self, db_adapter_type: Type[BaseAdapter], config):
     def submit_python_job(self, *args, **kwargs):
         return super().submit_python_job(*args, **kwargs)
 
+    @available
+    def db_materialization(self, context, materialization):
+        # NOTE: inspired by https://github.com/dbt-labs/dbt-core/blob/be4a91a0fe35a619587b7a0145e190690e3771c6/core/dbt/task/run.py#L254-L290
+        manifest = ManifestLoader.get_full_manifest(self.config)
```

Comment:
ATL (or a future TODO): should we consider caching the manifests? E.g. if you have tons of models, we would be generating a new manifest for each one of them (but IIRC one should suffice).